### PR TITLE
Android: Initial implementation of Interface submenu

### PR DESCRIPTION
**Pull request by Ebola16**
Fixes https://bugs.dolphin-emu.org/issues/10787
Better implementation of #6493.

General submenu is untouched. Things can be moved to different submenus as they are made. I had to rename some things for consistency to avoid something about "interface" being a protected system name.
Screenshots for @MayImilae 
screenshot_20180323-003420
screenshot_20180323-114722

### CODE REVIEW

**Comment by MayImilae** [comment]
Much better!

File: Source/Core/Core/Config/MainSettings.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ const ConfigInfo<int> MAIN_AUDIO_LATENCY{{System::Main, "Core", "AudioLatency"},
 const ConfigInfo<bool> MAIN_AUDIO_STRETCH{{System::Main, "Core", "AudioStretch"}, false};
 const ConfigInfo<int> MAIN_AUDIO_STRETCH_LATENCY{{System::Main, "Core", "AudioStretchMaxLatency"},
                                                  80};
+const ConfigInfo<bool> MAIN_USE_PANIC_HANDLERS{{System::Main, "Core", "UsePanicHandlers"}, true};
```
Comment (JosJuice) [inline Source/Core/Core/Config/MainSettings.cpp]: This still shouldn't be added.

Comment (Ebola16) [reply Source/Core/Core/Config/MainSettings.cpp]: Done

**Comment by Ebola16** [comment]
See images, added "Show OSD Messages" toggle. I wanted to add "Confirm on Stop" too but the INI setting does nothing. That'll be for a later PR.

**Comment by Ebola16** [comment]
@dolphin-emu-bot rebuild

**Comment by BhaaLseN** [comment]
Slightly off-topic here, but since I don't see you on IRC I'll just put it here anyways: I don't really use Dolphin that much on Android, so I had a hard time earlier setting up the on-screen controls (mostly because I had no idea about the pull-down menu). Any thoughts on adding a button in here somewhere (either Interface, since it kinda fits in there; or next to GC/Wii Controls where I was initially looking) to bring up that configuration screen without running a game?

**Comment by Ebola16** [comment]
Are you referring to the "Choose Controller" menu? If so I agree that it should be able to be accessed outside of emulation. I was thinking maybe above "GameCube Input"? If we like that location, I'd prefer to save that for a later PR.
screenshot_20180324-144652

**Comment by BhaaLseN** [comment]
Yeah, as I said; I just mentioned it here because I don't know how to reach you otherwise (and not because you should showhorn it into this PR). Thanks!

**Comment by Ebola16** [comment]
That's fine, I created [Android: Make "Choose Controller" menu accessible outside of emulation](https://bugs.dolphin-emu.org/issues/10962) for the future.

I'd like to see [Android: Choosing a different controller during emulation doesn't enable that controller](https://bugs.dolphin-emu.org/issues/10818) solved first though.

Thanks!

**Comment by Ebola16** [comment]
The last two seemingly unrelated commits are to give a consistent naming system for submenus.

**Comment by Ebola16** [comment]
Any update on this? I think it's ready for merge.

**Comment by leoetlino** [comment]
@Ebola16 it still needs somebody to review the Android code (which I'm not qualified to do).

**Comment by mahdihijazi** [comment]
@Ebola16 from the code point of view everything thing looks good to me but from the UI perspective I think we can come up with better order of settings and I need @MayImilae input on this. in #6338 I re-orderd the graphics settings section to have different sections and it sounds to me more appropriate place for UI settings like the one you added in this PR. This is the order I proposed in my PR:

screenshot

**Comment by Ebola16** [comment]
Sorry, I needed to document some annoying Dolphin Android bugs before returning to this.

@mahdihijazi I'm confused as to which settings you think should be moved to the graphics section. I've added "Use Panic Handlers" and "Show On-Screen Display Messages" in this PR for the interface submenu, which matches other Dolphin builds. The changes made to the graphics section are all typo fixes.

**Comment by Ebola16** [comment]
@mahdihijazi I think our comments are for the same goal but mine renames the strings that are involved with INIs. There's an interface submenu and an interface section in the INI and I wanted to differentiate between the two in strings.xml.

Technically this is a duplicated string but it helps with readability. Let me know if this is a poor design choice.

**Comment by Ebola16** [comment]
Anything holding this PR back? It's the base I'll be using to add other UI elements in the future.

**Comment by BhaaLseN** [comment]
You'll need to rebase it first on latest master, it cannot be merged as-is.

**Comment by Ebola16** [comment]
@BhaaLseN That must have happened quite recently, done.

**Comment by BhaaLseN** [comment]
Not really an Android guy, but the changes look ok-ish to me...except for: is there any reason why you truncated "description" to "descrip"? I'd rather have them readable (which is "description" if you ask me) than short. The only reason _against_ this is the number of occurrences (6 resources with "description" vs. 30 resources with "descrip"). I can accept the consistency-argument here for sticking with "descrip" though, unless someone else feels the same way.

**Comment by Ebola16** [comment]
I don't mind either way as long as it's consistent.

**Comment by leoetlino** [comment]
Yeah, any chance you could change them all to 'description'? 'descrip' just looks a bit weird and isn't that much shorter.

**Comment by Ebola16** [comment]
Sorry for some reason I missed the email for the update on this issue. I regex'd `discrip\b` to `description` so hopefully that covers all cases. @BhaaLseN @leoetlino

**Pull request merged**